benchmark/benchmark.go

```diff
@@ -30,3 +56,101 @@ func GetSubject() (string, error) {
 	}
 	return subject, nil
 }
+
+// Wrapper reports the benchmarking metrics of targetFunc to testing.B.
+func Wrapper(b *testing.B, targetFunc func(), cleanupFunc func()) {
+	metricsSum := Metrics{}
+	b.ResetTimer()
+	for n := 0; n < b.N; n++ {
+		b.StartTimer()
+		metrics, err := measureMetrics(targetFunc)
+		if err != nil {
+			b.Fatal("Error measuring metrics:", err)
+			return
+		}
+		b.StopTimer()
+		metricsSum.Add(metrics)
+		cleanupFunc()
+	}
+	b.ReportMetric(metricsSum.PeakCPUUsage/float64(b.N), "%cpu_peak/op")
+	b.ReportMetric(metricsSum.AverageCPUUsage/float64(b.N), "%cpu_avg/op")
+	b.ReportMetric(metricsSum.TotalCPUTime.Seconds()/float64(b.N), "cpu_seconds/op")
+	b.ReportMetric(float64(metricsSum.DiskUsageDelta/int64(b.N)), "disk_bytes/op")
+}
+
+func measureMetrics(f func()) (Metrics, error) { //nolint:unparam // make it extensible for future error handling
```

Comment:
nit: What is the responsibility of this `f()`? Can we give it a better name from which tells what is it for?